google/cloud/internal/async_read_write_stream_tracing.h

```diff
@@ -44,14 +44,18 @@ class AsyncStreamingReadWriteRpcTracing
   }
 
   future<bool> Start() override {
-    return impl_->Start().then([this](future<bool> f) {
-      auto started = f.get();
-      span_->SetAttribute("gl-cpp.stream_started", started);
-      return started;
-    });
+    auto start_span = internal::MakeSpan("Start");
+    return impl_->Start().then(
+        [this, ss = std::move(start_span)](future<bool> f) {
+          EndSpan(*ss);
+          auto started = f.get();
+          span_->SetAttribute("gl-cpp.stream_started", started);
```

Comment:
Ugh, so I feel like this would be more in line with OTel:

```cc
auto started = f.get();
ss->SetStatus(started
    ? opentelemetry::trace::StatusCode::kOk
    : opentelemetry::trace::StatusCode::kError);
ss->End()
return started;
```

.... but it does not look like Cloud Trace cares about that field. :unamused: 

I have seen it render things as red errors before, but that seems to be when HTTP status code attributes != success

![image](https://github.com/googleapis/google-cloud-cpp/assets/23088558/9ffa8acd-f703-43b1-bcb0-e73d7a1ba0c5)

---

Feel free to leave this alone.

Reply:
Ack.  I am not sure we should set the status there.  The actual error, if any, will be reported in `Finish()`.